apps/schema/src/components/NomnomDiagram.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div>
+    <InputCheckbox
+      id="displaySettings"
+      v-model="displaySettings"
+      :defaultValue="displaySettings"
+      :options="['attributes', 'external', 'vertical', 'ontologies']"
+    />
+    <div v-html="nomnomlSVG" class="bg-white" style="max-width: 100%"></div>
+  </div>
+</template>
+
+<script>
+import { renderSvg } from "nomnoml";
+import { InputCheckbox } from "molgenis-components";
+
+export default {
+  props: {
+    schema: {
+      type: Object,
+      required: true,
+    },
+  },
+  components: {
+    InputCheckbox,
+  },
+  data() {
+    return {
+      imgFullscreen: false,
+      displaySettings: [],
+    };
+  },
+  methods: {
+    nomnomColumnsForTable(table, tableName) {
+      let result = "";
+      if (
+        Array.isArray(table.columns) &&
+        this.displaySettings.includes("attributes")
+      ) {
+        result += "|";
+        table.columns
+          .filter((column) => column.table === tableName)
+          .forEach((column) => {
+            if (
+              column.columnType.includes("REF") ||
+              column.columnType.includes("ONTOLOGY")
+            ) {
+              result += `${column.name}: ${column.columnType.toLowerCase()}(${
+                column.refTable
+              })`;
+            } else {
+              result += `${column.name}: ${column.columnType.toLowerCase()}`;
+            }
+            result += `${column.nullable ? ";" : "*;"}`;
+          });
+        //remove trailing ;
+        result = result.replace(/;\s*$/, "");
+      }
+      return result;
+    },
+  },
+  computed: {
+    tables() {
+      return this.schema.tables;
+    },
+    ontologies() {
+      return this.schema.ontologies;
+    },
+    nomnomlSVG() {
+      return renderSvg(this.nomnomSource);
+    },
+    nomnomSource() {
```

Comment:
split up mega string build function into named function, nomnomSource does not describe the function

Reply:
what is added value of that? In terms on how we spend our time I feel it is wasteful to invest in tests for 'nomnom' as compared to many other features we can spend our time on.

As I wrote above: nomnom is a 'nice to have' and if there are errors this is not essential.